src/compiler/checker.ts

```diff
@@ -30279,7 +30307,16 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             else if (t.flags & TypeFlags.StructuredType) {
                 const prop = getPropertyOfType(t, name);
                 if (prop) {
-                    return isCircularMappedProperty(prop) ? undefined : removeMissingType(getTypeOfSymbol(prop), !!(prop && prop.flags & SymbolFlags.Optional));
+                    if (isCircularMappedProperty(prop)) {
+                        return undefined;
+                    }
+                    const type = getTypeOfSymbol(prop);
+                    if (type !== nonInferrableUnknownType) {
+                        return removeMissingType(type, !!(prop && prop.flags & SymbolFlags.Optional));
+                    }
+                    if (t.flags & TypeFlags.Intersection) {
+                        t = getIntersectionType((t as IntersectionType).types.filter(t => !t.pattern));
+                    }

```

Comment:
I wish I knew how to do it in a better way... this is mainly to satisfy the `isTupleType(t)` below (when it should be satisfied)